docs/tutorial/where-to-store-app-data.md

```diff
@@ -0,0 +1,93 @@
+
+On Windows, both `localUserData` and `userData` can be used to store persistent
+user data, the difference between them is, the `userData` can be synchronized to
+cloud, but `localUserData` can not.
+
+The convention is, if the user data takes lots of disk space, they should be
+written to `localUserData`, otherwise it can be written to `userData` instead.
+
+### `userCache`
+
+The `userCache` path is where cache data should be stored, files in this
+location should be safe to be removed without affecting app's state.
+
+On Linux it points to `$XDG_CACHE_HOME/AppName` or `~/.cache/AppName`.
+
+On macOS it points to `~/Library/Caches/AppName`.
+
+On Windows there is no specialized location to store cache data, and most apps
+write cache data to `%LOCALAPPDATA%`, so this path points to `localUserData` by
+default. If your app is writing files to both `localUserData` and `userCache`,
+you should consider changing the `userCache` path to a subdirectory of
+`localUserData` to avoid conflicts.
+
+## Data in Electron apps
+
+Data generated by a typical Electron app can be categorized to 3 kinds:
+
+1. Browsing data created by `Session` API, which in most apps are generated
+   implicitly when using the `BrowserWindow` API. They can be roughly separated
+   in 2 kinds:
+   1. HTTP disk cache and JavaScript code cache, which are written into the
+      `userCache` directory.
+   2. All other kinds of browsing data, such as cookies, localStorage, indexDB,
+      GPU Cache, and dictionaries. They are all considered as browsing data in
+      Chromium and written into the `sessionData` directory, which points to
+      `userData` by default.
+2. Data implicitly written by other Electron APIs, such as crash reports written
+   by the `crashReporter` module, or lock file written by the
+   `app.requestSingleInstanceLock` API. They are written to the `userData`
+   directory by default.
+3. Data generated by Electron apps themselves. It is developer's decision to
+   choose where to write them.
+
+## Choose a better location for sessionData
+
+For historical reasons the location of browsing data (`sessionData`) points to
+`userData` by default. This causes problems for apps that open remote websites
+with `BrowserWindow`, because disk spaces used by `sessionData` could increase
+quickly and users would get a very large `userData` directory, which is annoying
+for users who synchronizes content of `userData` to cloud.
+
+So for apps that don't store any user data inside browser, it is recommended
+to change `sessionData` to `userCache` to keep `userData` directory clean.
+
+```javascript
+app.setPath('sessionData', app.getPath('userCache'))
+```
+
+And for apps that rely on browser storage like cookies and localStorage to store
+user data, it is recommended to change `sessionData` to `localUserData`, to
+comply with user data conventions on Windows.
+
+```javascript
+app.setPath('sessionData', app.getPath('localUserData'))
+```
```

Comment:
I would start with the "apps that rely on browser storage" part before the "for apps that don't store any user data inside browser" part.